src/lib/services/batch-service.ts

```diff
@@ -186,30 +168,86 @@ export async function pollBatchOperation({
         if (state === 'Failed') {
           reject();
         } else if (state !== 'Running') {
-          resolve(parseInt(completeOperationCount, 10));
+          resolve(completeOperationCount);
         } else {
           setTimeout(() => {
             try {
               resolve(pollBatchOperation({ namespace, jobId }));
             } catch {
               reject();
             }
-          }, 1000);
+          }, 5000);
         }
       },
     );
   });
 }
 
-async function describeBatchOperation({
-  jobId,
-  namespace,
-}: DescribeBatchOperationOptions): Promise<BatchOperationInfo> {
+export async function describeBatchOperation(
+  { jobId, namespace }: DescribeBatchOperationOptions,
+  request = fetch,
+): Promise<BatchOperation> {
   const route = routeForApi('batch-operation.describe', {
     namespace,
   });
 
-  return requestFromAPI<BatchOperationInfo>(route, {
+  const response = await requestFromAPI<DescribeBatchOperationResponse>(route, {
     params: { jobId },
+    request,
+  });
+
+  return toBatchOperationDetails(response);
+}
+
+const toBatchOperationDetails = (
+  apiBatchOperationDetails: DescribeBatchOperationResponse,
+): BatchOperation => {
+  return {
+    ...apiBatchOperationDetails,
+    startTime: apiBatchOperationDetails.startTime,
+    closeTime: apiBatchOperationDetails.closeTime,
+    totalOperationCount: parseInt(
+      apiBatchOperationDetails.totalOperationCount,
+      10,
+    ),
+    completeOperationCount: parseInt(
+      apiBatchOperationDetails.completeOperationCount,
+      10,
+    ),
+    failureOperationCount: parseInt(
+      apiBatchOperationDetails.failureOperationCount,
+      10,
+    ),
+  };
+};
+
+export async function listBatchOperations(
```

Comment:
Just for my own education, do these ever go away after a certain period (retention period)? Or this call will return all batch operations of all time?

Reply:
That's a good question, I assume they do not adhere to the namespace's retention period, but I can double check on that.